Improve type handling for id params in generated swagger

djones6 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
<!--- Describe your changes in detail -->
Reworking of the handling of the id parameter, to permit both Int and String types rather than hard-coding `integer`.

This is code from @nhardman in #1304 which I have split into smaller PRs.

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->
This adds new Swagger generation tests for both String and Int type identifiers.

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] If applicable, I have updated the documentation accordingly.
- [x] If applicable, I have added tests to cover my changes.

ianpartridge commented on Tests/KituraTests/TestSwaggerGeneration.swift:
```diff
@@ -628,6 +633,69 @@ class TestSwaggerGeneration: KituraTest {
     }
 
+    func pathContentAssertions4(json: String?) {
+        if let jsonString = json {
```
`guard let` does exist ;)

ianpartridge commented on Sources/Kitura/SwaggerGenerator.swift:
```diff
@@ -316,11 +316,21 @@ struct SwaggerDocument: Encodable {
+            // id can only be string or integer. so force the type to string if
+            // it is neither.
+            var idType = "string"
+            if let ptype = SwiftType(rawValue: parametertype) {
```
I think I understand what's going on here, but it would be nice to make a few improvements:

1. `parametertype` -> `parameterType`
2. `swaggerType()` should be a property not a function
3. If the name can only be String or Integer why not define that in an enum rather than passing Strings everywhere

djones6 commented:
@ianpartridge I've addressed the review comments, except for replacing passing of strings with an enum: this bit of code needs re-working back up to the Router level to pass types through rather than string interpolating the type names.  I'd rather do that in a separate PR.

djones6 commented:
Travis is having issues (https://t.co/IX0wg2yH0J) so our Linux builds seem to be failing frequently.

djones6 commented:
@ianpartridge I've backed out the review changes so that this + the follow-on PRs can merge cleanly.  I'll implement your suggestions in the final one.
